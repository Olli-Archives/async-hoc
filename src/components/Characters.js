import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

export default function Characters({ charactersArray }){
  const characters = charactersArray.map(character=>{
    const { name, status, species, image } = character;
    return <Character key={`${name}${status}`} name={name} status={status} species={species} image={image}/>;
  });
  return (
   <>
   {characters}
   </>
  );

}

Characters.propTypes = {
  charactersArray: PropTypes.array.isRequired
}
;
