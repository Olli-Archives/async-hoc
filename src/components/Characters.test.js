import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('Characters component', ()=>{
  const charactersArray = [{ name:'rick', status:'alive', species:'human', image:'sample-image' }];
  it('renders characters from an array or characters', ()=>{
    const wrapper = shallow(<Characters charactersArray={charactersArray}/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
