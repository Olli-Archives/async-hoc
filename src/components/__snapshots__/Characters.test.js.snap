// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component renders characters from an array or characters 1`] = `
<Fragment>
  <Character
    image="sample-image"
    key="rickalive"
    name="rick"
    species="human"
    status="alive"
  />
</Fragment>
`;
